framework/include/vectorpostprocessors/LineMaterialSamplerBase.h

```diff
@@ -179,6 +184,7 @@ LineMaterialSamplerBase<T>::execute()
       addSample(_q_point[qp], qp_proj_dist_along_line, values);
     }
   }
+  _fe_problem.clearActiveMaterialProperties(_tid);
```

Comment:
I mean: we could make an `ElementLoopingGeneralUserObject` base class that works similarly to our `ThreadedElementLoop` with virtual function callbacks users can fill in and there would be some ability to feed an element list down to the base-class as well.

Just depends on how many times this is being done in the applications... if it's only a couple then it's not worth a base-class.  If it's more than 4 or 5 I would say we should abstract it out so people aren't doing it themselves.